app/src/main/java/org/p2p/wallet/common/EncryptedSharedPreferences.kt

```diff
@@ -0,0 +1,255 @@
+        }
+    }
+
+    fun <T> saveObject(key: String, data: T) {
+        tryWithLog(key) {
+            val objectAsJson = gson.toJson(data)
+            val encodedData = keyStoreWrapper.encode(key, objectAsJson)
+            sharedPreferences.edit { putString(key, encodedData) }
+        }
+    }
+
+    /**
+     * !!! WARNING !!!
+     * DOESN'T WORK WITH CLASSES
+     */
+    @Deprecated(message = "DOESN'T WORK, use only with primitive types like String")
+    fun <T> saveObjectList(key: String, data: List<T>) {
+        saveObject(key, data)
+    }
+
+    @Throws(IllegalArgumentException::class)
+    fun <T : Any> getObject(key: String, type: KClass<T>): T? {
+        return tryWithLog(key) {
+            sharedPreferences.getString(key, null)
+                ?.let { keyStoreWrapper.decode(key, it) }
+                ?.let { gson.fromJson(it, type.java) }
+        }
+    }
+
+    /**
+     * !!! WARNING !!!
+     * DOESN'T WORK WITH CLASSES
+     */
+    @Deprecated(message = "DOESN'T WORK, use only with primitive types like String")
+    fun <T : Any> getObjectList(key: String): List<T> {
+        return tryWithLog(key) {
+            sharedPreferences.getString(key, null)
+                ?.let { keyStoreWrapper.decode(key, it) }
+                ?.let { gson.fromJson(it, object : TypeToken<List<T>>() {}.type) }
+                ?: emptyList()
+        }
+    }
+
+    fun saveBytes(key: String, data: ByteArray) {
+        tryWithLog(key) {
+            val string = Hex.encode(data)
+            val encodedData = keyStoreWrapper.encode(key, string)
+            sharedPreferences.edit { putString(key, encodedData) }
+        }
+    }
+
+    fun getBytes(key: String): ByteArray? {
+        return tryWithLog(key) {
+            sharedPreferences.getString(key, null)
+                ?.let { keyStoreWrapper.decode(key, it) }
+                ?.let { Hex.decode(it) }
+        }
+    }
+
+    fun putBoolean(key: String, value: Boolean) {
+        tryWithLog(key) {
+            // no need to encode boolean values
+            sharedPreferences.edit { putBoolean(key, value) }
+        }
+    }
+
+    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
+        return tryWithLog(key) {
+            sharedPreferences.getBoolean(key, defaultValue)
+        }
+    }
+
+    fun putLong(key: String, value: Long) {
+        tryWithLog(key) {
+            // no need to encode boolean values
+            sharedPreferences.edit { putLong(key, value) }
+        }
+    }
+
+    fun getLong(key: String, defaultValue: Long): Long {
+        return tryWithLog(key) {
+            sharedPreferences.getLong(key, defaultValue)
+        }
+    }
+
+    /**
+     * Encodes each element inside the set and saves it to the shared preferences.
+     * Set is unique by definition, so when elements are encoded - they are unique too.
+     */
+    fun putStringSet(key: String, value: Set<String>) {
```

Comment:
I think, we can use List here, cause preferences does not work with set, and Kotlin.Set is MutableCollection
so I don't see reasons to have this method

Reply:
string set is supported by preferences, you can check android.content.SharedPreferences#getStringSet